rholang/src/main/scala/coop/rchain/rholang/interpreter/normalize.scala

```diff
@@ -257,6 +257,76 @@ object ProcNormalizeMatcher {
         )
       }
 
+      case p: PInput => {
+        import scala.collection.JavaConverters._
+        // To handle the most common case where we can sort the binds because
+        // they're from different sources, Each channel read starts its free variables at 0.
+        // We check for overlap at the end before sorting.
+
+        // We split this into parts. First we process all the sources, then we process all the bindings.
```

Comment:
Nitpick/preference but I thought this comment wasn't useful.

```
        val (sources, thisLevelFree) = processSources(bindings)
        val receipts                 = ReceiveSortMatcher.preSortBinds(processBindings(sources))
```
The above code makes it obvious that it is split into parts. The above comment already makes the reason clear as to why you're splitting it into two: the free variable count wouldn't be reset each source if you looped through everything at once.

Reply:
It's definitely odd to have data that mixes Name (from ADT) and Channel (from AST). I think the comment is warranted.